include/zio/zio_defs.h

```diff
@@ -0,0 +1,510 @@
+	 *
+	 * Expressed as a degree/minute/second notation string.
+	 */
+	DEV_CHAN_POS_LATITUDE,
+
+	/**
+	 * Longitude in a geo-localisation system.
+	 *
+	 * Expressed as a degree/minute/second notation string.
+	 */
+	DEV_CHAN_POS_LONGITUDE,
+
+	/**
+	 * The altitude or 'height' of an object or point in relation to an
+	 * unspecified reference level (ground, sea-level, another object,
+	 * etc.)
+	 *
+	 * Expressed in meters.
+	 */
+	DEV_CHAN_POS_ALT,
+
+	/**
+	 * The altitude or 'height' of an object or point in reference to sea
+	 * level.
+	 *
+	 * Expressed in meters.
+	 */
+	DEV_CHAN_POS_ALT_SEALEVEL,
+
+	/**
+	 * The altitude or 'height' of an object or point in reference to the
+	 * ground immediately below the point of measurement.
+	 *
+	 * Expressed in meters.
+	 */
+	DEV_CHAN_POS_ALT_GND,
+
+	/**
+	 * Distance in a straight line (regardless of obstacles) between the
+	 * source of the measurement and a specific reference point.
+	 *
+	 * Expressed in meters (m).
+	 */
+	DEV_CHANNEL_POS_DIST,
+
+	/**
+	 * Distance in a straight line along the surface of an object (such as
+	 * a sphere) between the source of the measurement and a specific
+	 * reference point.
+	 *
+	 * Expressed in meters (m).
+	 */
+	DEV_CHANNEL_POS_DIST_SURF,
+
+	/**
+	 * Proximity to an external plane or object.
+	 *
+	 * Expressed in millimeters (mm).
+	 */
+	DEV_CHANNEL_POS_PROX,
+
+	/**
+	 * Encapsulates a raw NMEA sentence from a GPS-type device
+	 *
+	 * Expressed as a null-terminated string, with the exact string
+	 * contents determined by NMEA standard. NMEA sentences are generally
+	 * limited to 79 characters plus the null-termination character.
+	 */
+	DEV_CHAN_POS_NMEA,
+
+	/**
+	 * An Euler angle, which consists of the three familiar X/Y/Z
+	 * headings.
+	 *
+	 * Expressed as a 3-vector (XYZ) of degrees.
+	 */
+	DEV_CHAN_ORIENT_EULER,
+
+	/**
+	 * A quaternion, which is generally used to represent a value in
+	 * three dimensional space, avoiding the problem of gimbal lock
+	 * associated with 3-vector Euler angles.
+	 *
+	 * Quaternions consist of four elements: a 3-vector and a scalar.
+	 */
+	DEV_CHAN_ORIENT_QUAT,
+
+	/**
+	 * An indication of tilt in degrees on a single axis relative to an
+	 * unspecified reference plane.
+	 *
+	 * Expressed in degrees (+/-180).
+	 */
+	DEV_CHAN_ORIENT_TILT,
+
+
+	/**
+	 * Pressure, which is a measurement of the force applied
+	 * perpendicularly to a specific surface area.
+	 *
+	 * Expressed in hectopascal (hPa)
+	 */
+	DEV_CHANNEL_PRESS,
+
+	/**
+	 * Barometric (or 'atmospheric') pressure, which is a measurement of
+	 * the pressure within the atmosphere. Related to the concept of
+	 * 'hydrostatic pressure' caused by the weight of the air above the
+	 * point of measure.
+	 *
+	 * TODO: Would millibar (mb) be a better unit choice, or should this
+	 * remain consist with the more general pressure channel unit (hPa)?
+	 *
+	 * Expressed in hectopascal (hPa)
+	 */
+	DEV_CHANNEL_PRESS_BAROM,
+
```

Comment:
Here I would keep hPa as unit (which equals mbar btw), just to be consistent with the usual unit.
Would DEV_CHANNEL_PRESS_AMBIENT be a better name?

Reply:
Ambient might actually be a clearer name, yes.

Reply:
I would prefer `DEV_CHANNEL_PRESS_ABSOLUTE` and `DEV_CHANNEL_PRESS_ABSOLUTE`, and perhaps `DEV_CHANNEL_PRESS_GAUGE`.

Most pressure sensors, e.g. BME280 or MS5637, produce absolute atmospheric pressure, which would range from about 1068 hPa down to 314 hPa.  Others like SDP810 produce relative pressure in a differential application, such as output and return in an HVAC system, which is often on the order of 10-25 Pa (with precision down to cPa).

Gauge pressure would be like a tire pressure sensor: relative to absolute atmospheric pressure.

Terms like "barometric" or "atmospheric" may be interpreted as referring to absolute pressure adjusted for altitude, like you get in weather reports.  This can be very different from absolute pressure.  Barometric pressure could be a derived measurement, but having a sensor that measures it directly seems pretty unlikely (maybe one built into a GPS).  Avoiding those terms might reduce confusion.

As for units: I'd really prefer SI throughout, so Pa.  Let the consumer deal with scale conversions.